clients/python/agentic-sandbox-client/agentic_sandbox/sandbox_client.py

```diff
@@ -113,37 +121,50 @@ def _wait_for_sandbox_ready(self):
             field_selector=f"metadata.name={self.claim_name}",
             timeout_seconds=self.sandbox_ready_timeout
         ):
-            sandbox_object = event['object']
-            status = sandbox_object.get('status', {})
-            conditions = status.get('conditions', [])
-            is_ready = False
-            for cond in conditions:
-                if cond.get('type') == 'Ready' and cond.get('status') == 'True':
-                    is_ready = True
-                    break
+            if event["type"] in ["ADDED", "MODIFIED"]:
+                sandbox_object = event['object']
+                status = sandbox_object.get('status', {})
+                conditions = status.get('conditions', [])
+                is_ready = False
+                for cond in conditions:
+                    if cond.get('type') == 'Ready' and cond.get('status') == 'True':
+                        is_ready = True
```

Comment:
The logic here relies on the `pod-name` annotation being present when the `Ready` condition is `True`. Is there a chance the annotation could be added in a separate `MODIFIED` event after the sandbox is already `Ready`? This could create a race condition where the client proceeds with the wrong pod name. It's worth verifying if the controller guarantees the annotation is present when the sandbox is marked as ready.